src/Symfony/Component/Validator/Constraints/JwtValidator.php

```diff
@@ -0,0 +1,41 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Validator\Constraints;
+
+/**
+ * @author Mokhtar Tlili <tlili.mokhtar@gmail.com>
+ */
+class JwtValidator extends AbstractFormatValidator
+{
+    protected function validateFormat(string $value): bool
+    {
+        $parts = explode('.', $value);
+
+        // JWT should have exactly three parts
+        if (3 !== \count($parts)) {
+            return false;
+        }
+
+        [$header, $payload] = $parts;
+
+        // Replace URL-safe characters with standard Base64 characters and decode
+        $decodedHeader = base64_decode(strtr($header, '-_', '+/'));
+        $decodedPayload = base64_decode(strtr($payload, '-_', '+/'));
+
+        return json_validate($decodedHeader) && json_validate($decodedPayload);
```

Comment:
I don't think this is enough to validate that this is a valid JWT. Having a core validator doing a *partial* validation is worse than not having it IMO.

Reply:
Agree. Some may rely on a custom constraint to invoke `lcobucci/jwt` (for example) to ensure everything's in place, including verifying the token signature.

I understand that being able to validate a JWT with a constraint is nice, but I would expect a constraint like this to check signature/issuer/etc. (which would greatly complexity the logic in the validator), otherwise you would need to write a second checker anyway to do some asserts on info included in the token. This would then void the usefulness of such constraint.

Reply:
Thank you all for your valuable feedback. I will remove the JWT constraint for now and might push a separate PR for it later, as it requires extensive work to verify the signature.